detox/src/utils/environment.js

```diff
@@ -17,12 +18,19 @@ function getAndroidSDKPath() {
 }
 
 function getAndroidEmulatorPath() {
-  const sdkPath = getAndroidSDKPath();
-  const extension = os.platform() === 'win32' ? '.exe' : '';
-  const newEmulatorPath = path.join(sdkPath, 'emulator', `emulator${extension}`);
-  const oldEmulatorPath = path.join(sdkPath, 'tools', `emulator${extension}`);
-
-  return fs.existsSync(newEmulatorPath) ? newEmulatorPath : oldEmulatorPath;
+  try{
+    const extension = os.platform() === 'win32' ? '.exe' : '';
```

Comment:
I have an assumption, that you can do here much better, given that now you are leveraging `which` logic.

```js
return
  which.sync('emulator', { path: preferedPath, nothrow: true }) ||
  which.sync('emulator', { path: alternativePath, nothrow: true }) ||
  which.sync('emulator'); /* the latter will throw */
```

And no dances with `if fs.existsSync()` and `os.platform()`.

What do you think?